apps/docusaurus/docs/move/move-on-aptos/development/design-patterns.md

```diff
@@ -0,0 +1,188 @@
+is ideal to ensure the calling application checks the amount prior. The output
+will also always be the same, the `amount` will be minted.
+
+```move
+module deployer_address::error_example {
+  use mint_address::minter;
+
+  /// Requested to mint more than is left to mint
+  const E_MINT_AMOUNT_TOO_HIGH: u64 = 1;
+
+  /// Mint multiple NFTs, aborting if there are not enough
+  public fun mint_many(caller: &signer, amount: u64) {
+    // Cancel the transaction if there are too many to mint
+    assert!(amount <= minter::amount_left(), E_MINT_AMOUNT_TOO_HIGH);
+
+    // Mint the amount
+    for (i in 0..amount) {
+      minter::mint_one();
+    }
+  }
+}
+```
+
+### Handle conditions without abort
+
+The other pattern we can take is to never abort the transaction unless there is
+a particular reason to abort. This ensures state will always change regardless
+of error conditions.
+
+In the example below, the user mints up to the amount requested and it will not
+abort if there are zero NFTs to actually mint. This is ideal in that the calling
+application does not need to know the amount ahead of time. The caveat is that
+the output state is different, it can be any value from 0 minted up to `amount`
+minted of NFTs.
+
+```move
+module deployer_address::error_example {
+
+  use mint_address::minter;
+
+  /// Mint multiple NFTs up to the amount requested
+  /// This will never fail, even if there's none to mint
+  public fun mint_many(caller: &signer, amount: u64) {
+    for (i in 0..amount) {
+      // Mint as many as we can, if there are no more, quit
+      if (0 == minter::amount_left()) {
+        return
+      };
+
+      // Mint a single NFT
+      minter::mint_one();
+    }
+  }
+}
+```
+
+## Permissions
+
+Access control and permissions are handled specifically by the contract writer.
+If access control is skipped, it is possible that there is unexpected access to
+the contract, and can lead to bugs and unexpected behavior.
+
+### Access Control List (ACL) Model
+
+An access control list (ACL) specifically checks a list or known addresses to
+ensure that only known users can access (or not access) a function. This
+specifically must use a `signer` as an input, as proof that the user is who they
+say they are.
+
+```move
+module deployer_address::access_control_example {
+  use aptos_framework::signer;
+
+  /// User is not allowed
+  const E_NOT_ALLOWED: u64 = 1;
+  /// User is denied
+  const E_DENIED: u64 = 2;
+
+  struct AccessControl has key {
+    allowlist: vector<address>,
+    denylist: vector<address>
+  }
+
+  fun init_module(deployer: &signer) {
+    move_to(deployer, AccessControl {
+      allowlist: vector[],
+      denylist: vector[],
+    })
+  }
+
+  public fun only_allowed(caller: &signer) {
+    let caller_address = signer::address_of(caller);
+
+    let acl = borrow_global<AccessControl>(@deployer_address);
+    assert!(vector::contains(acl.allowlist), E_NOT_ALLOWED);
```

Comment:
This is missing the second argument for vector::contains. Also same thing, let's use error::something(E_SOMETHING).